### PR TITLE
Feat: add kompose client PoC

### LINKED ISSUE #464: using kompose as a library
Is it possible to use `kompose` as a go library, which can just be imported and used.

Something like `artifacts := kompose.Convert("/app1/docker-compose.yml", "kubernetes")`.

It would be very useful to be able to use kompose as a library, we should investigate that :)

CC: @kadel

### PR DESCRIPTION
**Pull request by AhmedGrati**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Implement a PoC that adds a Kompose client.
**Note: For now it's just a PoC. @cdrage and @hangyan your input is really valuable here.**

#### Which issue(s) this PR fixes:

Fixes #464.

### CODE REVIEW

**Review by cdrage — COMMENTED** [review]
@AhmedGrati The work you've done is absolutely excellent. I'm super excited to get this merged in one day because we've been missing a library / client for ages and it'd be wonderful to have.

One question though is why there are changes to the tests? Since we're just adding the client, I don't think anything should be changing in `script/test/fixtures` ?

We may need to rebase I believe.

I did test this out and it worked wonderfully :) I would like to go through this a bit more and create some documentation though, as well as a new section of the site so that people can use the library  like so.

**Comment by AhmedGrati** [comment]
@cdrage thanks a lot 🙏, I will go through another iteration in the upcoming days to remove these changes in the test files, rebase and add other missing options in the convert command.

**Comment by AhmedGrati** [comment]
@cdrage it should be good now. Can you take a look at it, please?

**Comment by cdrage** [comment]
Your code always surprises me! I honestly have no input.. the way that you structured the input parameters and the conversion method makes A LOT of sense and how you return the data as well.

All of this LGTM and thank you for this amazing (and big!) task.

We should open an issue to create documentation on how to actually use this.

You good if I make a release this week so people can use the library?

File: client/convert.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+		ServiceGroupName:            k.serviceGroupName(options),
+		SecretsAsFiles:              k.secretsAsFiles(options),
+		GenerateNetworkPolicies:     options.GenerateNetworkPolicies,
+	}
+	err = app.ValidateComposeFile(&kobjectConvertOptions)
+	if err != nil {
+		return nil, err
+	}
+	objects, err := app.Convert(kobjectConvertOptions)
+	return objects, err
```
Comment (cdrage) [inline client/convert.go:58]: I like the fact you return the object and then the user can convert that to yaml / json with their own library, as well as the error output. Good job!

**Review by cdrage — APPROVED** [review]

**Comment by AhmedGrati** [comment]
@cdrage thanks a lot 😊😊 Yeah We can make a release this week. Should we merge this PR?

**Comment by cdrage** [comment]
Yup! If you can rebase + make sure tests pass (since other PR's got merged in) as soon as it's green, merge it in!

**Comment by realgam3** [comment]
That's an amazing PR!
I wrote my tool [kompose-ex](https://github.com/realgam3/kompose-ex) in Python because I didn't had this... 
I'm waiting for this 🙏

**Pull request merged**